src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -813,6 +815,13 @@ protected function asDateTime($value)
         return Date::createFromFormat($format, $value);
     }
 
+    protected function asTime($value)
+    {
+        return $value instanceof Carbon || $value instanceof CarbonInterface
```

Comment:
Isn't `$value instanceof Carbon` redundant? `Illuminate\Support\Carbon` extends `\Carbon\Carbon` implements `\Carbon\CarbonInterface`

Reply:
Yes it is. But I decided to do by analogy to `asDateTime()` method:
```php
protected function asDateTime($value)
{
    // If this value is already a Carbon instance, we shall just return it as is.
    // This prevents us having to re-instantiate a Carbon instance when we know
    // it already is one, which wouldn't be fulfilled by the DateTime check.
    if ($value instanceof Carbon || $value instanceof CarbonInterface) {
        return Date::instance($value);
    }
```

Reply:
Removed overhead